lib/krane/deploy_task.rb

```diff
@@ -277,16 +277,42 @@ def validate_configuration(prune:)
     end
     measure_method(:validate_configuration)
 
+    def partition_dry_run_resources(resources)
+      individuals = []
+      mutating_webhooks = cluster_resource_discoverer.fetch_mutating_webhook_configurations
+      mutating_webhooks.each do |spec|
+        spec.dig('webhooks').each do |webhook|
+          match_policy = webhook.dig('matchPolicy')
+          webhook.dig('rules').each do |rule|
+            next if %w(None NoneOnDryRun).include?(rule.dig('sideEffects'))
+            groups = rule.dig('apiGroups')
+            versions = rule.dig('apiVersions')
+            kinds = rule.dig('resources').map(&:singularize)
+            groups.each do |group|
+              versions.each do |version|
+                kinds.each do |kind|
+                  individuals += resources.select do |r|
+                    (r.group == group || group == '*' || match_policy == "Equivalent") &&
+                    (r.version == version || version == '*' || match_policy == "Equivalent") &&
+                    (r.type.downcase == kind.downcase)
+                  end
+                  resources -= individuals
```

Comment:
I can never remember the pass by semantics of ruby but, Is resources a copy or does this impact `validate_resources`

Reply:
It's pass by reference, however the `-=` operator returns a new object, not a mutation of the old one. If we used, e.g. `<<`, then we would mutate the incoming object. As a matter of caution, I'll `dup` the `resources` array to avoid any issues in the future.